proxy/src/transport/tls/cert_resolver.rs

```diff
@@ -0,0 +1,131 @@
+use std::{
+    sync::Arc,
+    time::SystemTime,
+};
```

Comment:
cert_resolver.rs contains logic that will be moved to Rustls and/or webpki once I can change the Rustls API. This implementation lets us check the certificate for validity (which Rustls doesn't do) and lets us use ECDSA certificates (which are faster for the server to use, and also much easier and faster to generate private keys for in the RA/CA component).